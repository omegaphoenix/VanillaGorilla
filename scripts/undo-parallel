#!/bin/bash

rm -r datafiles/
./nanodb < schemas/writeahead/create_t.sql > /dev/null

./nanodb < schemas/writeahead/transaction1.sql > scripts/output/t1.out &
./nanodb < schemas/writeahead/transaction2.sql > scripts/output/t2.out
./nanodb < schemas/writeahead/read_t.sql | grep -v "main" | grep '+\||' > scripts/output/undo-parallel.out

if [ -z $(diff scripts/output/undo-parallel.out scripts/output/undo-parallel.expected) ] 
then
    echo "Test undo-parallel passed!"
else
    echo "Test undo-parallel FAILED."
fi
