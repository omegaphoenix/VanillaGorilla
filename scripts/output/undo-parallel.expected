+----+
| A  |
+----+
|  1 |
|  2 |
| 30 |
| 40 |
+----+
